aws-cfn/cross-service/resilient-workflow/README.md

```diff
@@ -0,0 +1,240 @@
+
+3. **Static response** — Updates a parameter that prompts the web server to return a static response when the
+   recommendation service fails. Requests for recommendations now return a static response,
+   which is a better customer experience.
+
+   Edit `params.json`. Change the entry with `ParameterKey` as `SSMFailure` and set `ParameterValue` to `static`.
+
+4. **Bad credentials** — Sets the table name parameter so the recommendations service succeeds, but also
+   updates one of the instances to use an instance profile that contains bad credentials.
+   Now, when the load balancer selects the bad instance to serve a request, it returns
+   a static response because it cannot access the recommendation service, but the other
+   instances return real recommendations.
+
+   Edit `params.json` and remove the `SSMTable` entry from step 2. Edit `resilient-service.yaml`. Update the ... to use ....
+
+5. **Deep health checks** — Sets a parameter that instructs the web server to perform a deep health check.
+   For this example, a deep health check means that the web server reports itself as unhealthy if it can't
+   access the recommendations service. The instance with bad credentials reports as unhealthy and the load
+   balancer takes it out of rotation. Now, requests are forward only to healthy instances.
+
+   Edit `params.json`. Change the entry with `ParameterKey` as `SSMHealthCheck` and set `ParameterValue` to `deep`.
+
+6. **Replace the failing instance** — Terminates the unhealthy instance and lets Amazon EC2 Auto Scaling start
+   a new instance in its place. During this process, the stopping and starting instances are unhealthy so they
+   don't receive any requests, but the load balancer continues to forward requests to healthy
+   instances. When the new instance is ready, it is added to the rotation and starts receiving
+   requests.
+
+   Using the AWS Console, open the CloudFormation page. Navigate to the `resilience-demo` stack. Click on the `Resources`
+   tab. Find the `DocExampleRecommendationServiceTargetGroup` line. Click on the link to the "Physical Resource ID".
+   From this EC2 page, find the list of instances in the target group. Select one and navigate to it. Choose Actions ->
+   Terminate instance. See EC2 terminate the instance, and watch the Autoscaling Group start a new instance.
+
+7. **Fail open** — Sets the table name parameter so the recommendations service fails for all instances.
+   Because all instances are using deep health checks, they all report as unhealthy. In this
+   case, the load balancer continues to forward requests to all instances. This lets the
+   system fail open and lets the instances return static responses, rather than fail closed
+   and report failure.
+
+   Edit `params.json`. Add a new entry with `ParameterKey` as `SSMTable` and `ParameterValue` as `bad_table`.
+   After updating, the service should report healthy but return `500` responses.
```

Comment:
This should be the opposite. Health checks should be unhealthy, but the load balancer should forward requests to them anyway and because we previously implemented static response, it should return a static response.

Reply:
Done.